scheduler/src/cook/kubernetes/api.clj

```diff
@@ -69,6 +70,28 @@
   [pod-name]
   (str/starts-with? pod-name cook-synthetic-pod-name-prefix))
 
+(defn pod-name->job-uuid
+  "If a pod is synthetic, return the uuid of the job it was created for"
+  [pod-name]
+  (when (synthetic-pod? pod-name)
+    (re-find #"[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}" pod-name)))
+
+(defn pod-name->instance-uuid
+  "If a pod is not synthetic, return the instance id associated with its pod-name"
+  [pod-name]
+  (when-not (synthetic-pod? pod-name)
+    pod-name))
+
+(defn create-uuid-map
+  "Return a map with :job-uuid and/or :instance-uuid based on pod-name"
```

Comment:
It will currently never `assoc` both, right?

That being said, we probably eventually want to have the job uuid everywhere that we have the instance uuid.

Reply:
Yeah, currently it will never assoc both, I just said "and/or" because at some point we might want both. I can just change it to "or" to be more exact if that's preferable.

And yeah, eventually we do want to have job uuid (and will have it). But it looks like that will involve a more involved refactor and we think it's out of scope for this PR right now.